src/AbstractNav.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import qsa from 'dom-helpers/query/querySelectorAll';
+import PropTypes from 'prop-types';
+import elementType from 'prop-types-extra/lib/elementType';
+import mapContextToProps from 'react-context-toolbox/lib/mapContextToProps';
+import SelectableContext, { makeEventKey } from './SelectableContext';
+import NavContext from './NavContext';
+import TabContext from './TabContext';
+
+const noop = () => {};
+
+class AbstractNav extends React.Component {
```

Comment:
cheeeck this out @taion i quite like the sort of composability of the new contexts